packages/frontend/src/utils/axios.ts

```diff
@@ -9,10 +10,48 @@ const isJsonBlob = (data): data is Blob =>
  *
  * @param err
  */
-export const handleErrors = (
-  err: AxiosError,
+export const handleErrors = async (
+  err: AxiosError<{
+    code: number;
+    message: string;
+    statusCode: number;
+  }>,
   handleErrorsAutomatically = true
-): AxiosError => {
+): Promise<
+  AxiosError<{
+    code: number;
+    message: string;
+    statusCode: number;
+  }>
+> => {
+  let refreshToken;
+  const recoilPersist = localStorage.getItem('recoil-persist');
+  if (recoilPersist) {
+    refreshToken = JSON.parse(recoilPersist)?.ActiveTokenSet?.refreshToken;
+  }
+
+  if (
+    refreshToken &&
+    recoilPersist &&
+    err?.response?.status === 401 &&
+    // 1092, 1107 are the error codes for invalid jwt token that defined in backend
+    (err?.response?.data?.code === 1092 || err?.response?.data?.code === 1107)
```

Comment:
1092 means unauthenticated user trying to access authenticated resource. This should not trigger a refresh.